test/cloud-sql-instance.ts

```diff
@@ -100,52 +123,140 @@ t.test('CloudSQLInstance', async t => {
       authType: AuthTypes.PASSWORD,
       instanceConnectionName: 'my-project:us-east1:my-instance',
       sqlAdminFetcher: fetcher,
+      limitRateInterval: 50,
     });
-    const refreshFn = instance.refresh;
     instance.refresh = () => {
       if (refreshCount === 2) {
-        instance.cancelRefresh();
         const end = Date.now();
         const duration = end - start;
         t.ok(
           duration >= 100,
           `should respect refresh delay time, ${duration}ms elapsed`
         );
+        instance.cancelRefresh();
         return t.end();
       }
       refreshCount++;
       t.ok(refreshCount, `should refresh ${refreshCount} times`);
-      refreshFn.call(instance);
+      CloudSQLInstance.prototype.refresh.call(instance);
     };
     // starts out refresh logic
     instance.refresh();
   });
 
+  t.test('refresh error', async t => {
```

Comment:
small nit: this description doesn't describe the desired behavior -- what do we do when we hit a refresh error? Not throw an exception?

Reply:
ditto below